src/pages/Flavors.js

```diff
@@ -1,198 +1,174 @@
 import { PagerInfo, withPagination } from 'components/Pagination/Pagination';
 import { Col, Container, Row, Table } from 'react-bootstrap';
 import ToggleButton from 'components/ToggleButton/ToggleButton';
-import { actions as appActions } from 'reducers/application';
 import { actions as flavorActions } from 'reducers/flavor';
 import { actions as flavorsActions } from 'reducers/flavors';
 import { isLoggedIn } from 'selectors/application';
 import { getStash, isLoaded } from 'selectors/flavor';
 import { getAllFlavors, getFlavorsPager } from 'selectors/flavors';
 
-export class Flavors extends Component {
-  static propTypes = {
-    actions: PropTypes.object.isRequired,
-    collection: PropTypes.array.isRequired,
-    loggedIn: PropTypes.bool.isRequired,
-    pager: PropTypes.object.isRequired,
-    pagerNavigation: PropTypes.node.isRequired,
-    stash: PropTypes.array,
-    stashLoaded: PropTypes.bool.isRequired
-  };
+function StashToggle({ value, onClick }) {
+  return (
+    <ToggleButton
+      data-testid="stash-toggle"
+      value={value}
+      onClick={onClick}
+      title={!value ? 'Enable Flavor Stash' : 'Disable Flavor Stash'}
+      variant="switch"
+    />
+  );
+}
 
-  constructor(props) {
-    super(props);
+StashToggle.propTypes = {
+  value: PropTypes.bool.isRequired,
+  onClick: PropTypes.func.isRequired
+};
+
+function StashIcon({ id, has, onAdd, onRemove }) {
+  return (
+    <ToggleButton
+      data-testid={`stash-icon-${id}`}
+      value={has}
+      onClick={() => (has ? onRemove(id) : onAdd(id))}
+      title={has ? 'Remove from Stash' : 'Add to Stash'}
+      variant="check"
+    />
+  );
+}
 
-    this.state = { stashToggle: false };
+StashIcon.propTypes = {
+  id: PropTypes.string.isRequired,
+  has: PropTypes.bool.isRequired,
+  onAdd: PropTypes.func.isRequired,
+  onRemove: PropTypes.func.isRequired
+};
 
-    const { loggedIn } = this.props;
+export function Flavors({ collection, pager, pagerNavigation }) {
+  const dispatch = useDispatch();
 
-    if (loggedIn) {
-      this.handleStashToggle = this.handleStashToggle.bind(this);
-      this.handleAddToStash = this.addToStash.bind(this);
-      this.handleRemoveFromStash = this.removeFromStash.bind(this);
-    }
-  }
+  const loggedIn = useSelector(isLoggedIn);
+  const stash = useSelector(getStash);
+  const stashLoaded = useSelector(isLoaded);
 
-  componentDidMount() {
-    const { actions, loggedIn, stashLoaded } = this.props;
+  const [stashToggle, setStashToggle] = useState(false);
+  const [holdings, setHoldings] = useState({});
 
-    if (loggedIn) {
-      if (!stashLoaded) {
-        actions.requestStash();
-      }
+  useEffect(() => {
+    if (loggedIn && !stashLoaded) {
+      dispatch(flavorActions.requestStash());
     }
-  }
-
-  handleStashToggle() {
-    const { stash } = this.props;
-    const { holdings: stashMap } = this.state;
-    const holdings = {};
+  }, [dispatch, loggedIn, stashLoaded]);
```

Comment:
```
[dispatch, loggedIn, stashLoaded]
```
there's a performance issue with including `stashLoaded` in this list, but the eslint rules force me to do it. The behavior of the original component doesn't match this because the effect will be ran again every time stashLoaded set to false which happens on every stash update action in the reducer.

Reply:
Yeah, adding a comment and disabling the rule for that line sounds fine to me.